src/Microsoft.DotNet.Darc/DarcLib/LocalLibGit2Client.cs

```diff
@@ -643,4 +643,29 @@ public Task<List<GitTreeItem>> LsTreeAsync(string repoPath, string gitRef, strin
 
         return Task.FromResult(gitTreeItems);
     }
+
+    public string GetMergeBaseCommitSha(string repoPath, string branchOrCommit1, string branchOrCommit2)
+    {
+        if (string.IsNullOrWhiteSpace(repoPath))
+            throw new ArgumentException("Repository path must be provided.", nameof(repoPath));
+        if (string.IsNullOrWhiteSpace(branchOrCommit1))
+            throw new ArgumentException("First branch or commit must be provided.", nameof(branchOrCommit1));
+        if (string.IsNullOrWhiteSpace(branchOrCommit2))
+            throw new ArgumentException("Second branch or commit must be provided.", nameof(branchOrCommit2));
+
+        using var repo = new Repository(repoPath);
```

Comment:
We try to not use usages of `LibGit2Sharp` as it has a massive memory hit (more so when using it with the VMR). I believe we already compute merge bases and it's a single git command via the process manager so it might be better to implement this inside of `LocalGitClient` and `ILocalGitClient` than here:

```csharp
ProcessExecutionResult result = await repo.ExecuteGitCommand("merge-base", gitRef1, gitRef2);
result.ThrowIfFailed($"Failed to find a common ancestor for {gitRef1} and {gitRef2}");
return result.GetOutputLines().First();
```